src/calliope/model.py

```diff
@@ -188,9 +188,6 @@ def _init_from_model_def_dict(
         model_config.union(model_definition.pop("config"), allow_override=True)
 
         init_config = update_then_validate_config("init", model_config)
-        # We won't store `init` in `self.config`, so we pop it out now.
```

Comment:
I am now allowing `config.init` to remain in `model._model_data.attrs`.
I think it makes the model creation more transparent, particularly because the added math files specified are kept.